connector/connect/src/main/protobuf/spark/connect/relations.proto

```diff
@@ -250,3 +251,15 @@ message SubqueryAlias {
   // Optional. Qualifier of the alias.
   repeated string qualifier = 3;
 }
+
+// Rename columns on the input relation.
+message RenameColumns {
```

Comment:
We definitely need this proto message, and we can probably add a corresponding logical if https://github.com/apache/spark/pull/38395 is the right direction.

cc @HyukjinKwon

Reply:
i am afraid we have to add a logical for nearly each df API, even for simple functions like `toDF` ( as well as `withColumn`, [`dropDuplicates`](https://github.com/apache/spark/commit/b14da8b1b65d9f00f49fab87f738715089bc43e8)) which depend on analyzed plan.

Reply:
I think as of now we are leaning to not adding new catalyst plan.